cognite/extractorutils/rest/extractor.py

```diff
@@ -546,25 +546,28 @@ def inner_call() -> Response:
             resp.raise_for_status()
             return resp
 
-        raw_response = inner_call()
-
-        try:
-            print(endpoint.endpoint.response_type)
-            if endpoint.endpoint.response_type == JsonBody:
-                response = raw_response.json()
-            elif endpoint.endpoint.response_type == Response:
-                response = raw_response
-            else:
-                data = raw_response.json()
-                if isinstance(data, list):
-                    data = {"items": data}
-                response = dacite.from_dict(endpoint.endpoint.response_type, data)
-
-            result = endpoint.endpoint.implementation(response)
-            self.handle_output(result)
-        except (JSONDecodeError, DaciteError) as e:
-            self.logger.error(f"Error while parsing response: {str(e)}")
-            raise e
+        raw_response: Response = inner_call()
+
+        if raw_response.status_code == HTTPStatus.NO_CONTENT:
```

Comment:
Not sure I agree with having this here, if the response type is set to `Response`, then we should return the full response object, independent of status code.

Reply:
That's if you are asking for the Response.

What this is fixing is if you are looking for a data response, aka the `else` condition, but the API you queried from gave a NO_CONTENT (204) response. This particular data response does not have `.json()`, thus it would error and kill the extractor.

Thoughts on improvement?

Reply:
I understand, my point is that with your change, if you set response type to `Response`, then receive a `NO_CONTENT`, you will receive an empty dictionary, instead of the full response object, which is not right.

Instead move the status code check into where we call `.json()`

Reply:
Ahh. I gotcha.

Well, it does fix my issue, if I do that... would you expect the issue to still persist if response type was set to `JsonBody`?

Should I also put the `NO_CONTENT` check in that condition?

Reply:
Yeah, it ends up being a little clunky. Maybe we should refactor to something like

```
if endpoint.endpoint.response_type == Response:
    ...
else:
    # Your check here
    if endpoint.endpoint.response_type == JsonBody:
         ...
    else:
        ...
```

or whatever. I do think we should check for NO_CONTENT whenever we call `.json()`, it makes sense.

Reply:
Sounds good. Let me change that as you suggested.